docs/import/import_data.js

```diff
@@ -140,132 +135,327 @@ function stringifyGeoid(feature) {
+  }
+  const incomeKey = assetData['income_key'];
+  if (!incomeKey) {
+    return missingAssetError('column name for income table');
+  }
+  // If we switch to CrowdAI data, this will change.
+  const buildingPaths = assetData['building_asset_paths'];
+  if (!buildingPaths) {
+    return missingAssetError('building data asset paths');
+  }
+  const {damage, damageEnvelope} = calculateDamage(assetData);
+  if (!damageEnvelope) {
+    // Must have been an error.
+    return false;
+  }
+
+  let allStatesProcessing = ee.FeatureCollection([]);
+  for (const state of states) {
+    const snapPath = snapPaths[state];
+    if (!snapPath) {
+      return missingAssetError('SNAP asset path for ' + state);
+    }
+    const sviPath = sviPaths[state];
+    if (!sviPath) {
+      return missingAssetError('SVI asset path for ' + state);
+    }
+    const incomePath = incomePaths[state];
+    if (!incomePath) {
+      return missingAssetError('income asset path for ' + state);
+    }
+    const buildingPath = buildingPaths[state];
+    if (!buildingPath) {
+      return missingAssetError('building asset path for ' + state);
+    }
+    const blockGroupPath = blockGroupPaths[state];
+    if (!blockGroupPath) {
+      return missingAssetError(
+          'Census TIGER block group shapefile for ' + state);
+    }
+
+    const stateGroups =
+        ee.FeatureCollection(blockGroupPath).filterBounds(damageEnvelope);
+
+    let processing = ee.FeatureCollection(snapPath).map(stringifyGeoid);
+
+    // Join snap stats to block group geometries.
+    processing =
+        innerJoin(processing, stateGroups, censusGeoidKey, tigerGeoidKey);
+    processing = processing.map((f) => combineWithSnap(f, snapKey, totalKey));
+    // Join with income.
+    // TODO: make income formatting prettier so it looks like a currency value.
+    //  Not trivial because it has some non-valid values like '-'.
+    processing = innerJoin(processing, incomePath, geoidTag, censusGeoidKey);
+    processing =
+        processing.map((f) => combineWithAsset(f, incomeTag, incomeKey));
+    // Join with SVI (data is at the tract level).
+    processing = processing.map(addTractInfo);
+
+    processing = innerJoin(processing, sviPath, tractTag, cdcGeoidKey);
+    processing = processing.map((f) => combineWithAsset(f, sviTag, sviKey));
+
+    // Get building count by block group.
+    const buildingsHisto =
+        computeBuildingsHisto(damageEnvelope, buildingPath, state, stateGroups);
+
+    // Create final feature collection.
+    processing = processing.map(
+        (f) => countDamageAndBuildings(f, damage, buildingsHisto));
+    allStatesProcessing = allStatesProcessing.merge(processing);
+  }
+
+  const assetName = 'data-ms-as-tot';
+  const scoreAssetPath = gdEePathPrefix + getDisaster() + '/' + assetName;
+  try {
+    ee.data.deleteAsset(scoreAssetPath);
+  } catch (err) {
+    if (err.message !== 'Asset not found.') {
+      throw err;
```

Comment:
Do we want to throw an error when there isn't an old asset to delete? What about the first time running this script?

Reply:
We're throwing an error only if the error is *not* "Asset not found". Rearranged to make clearer.